Revit_Core_Engine/Query/LookupParameter.cs

```diff
@@ -62,6 +62,31 @@ public static Parameter LookupParameter(this Element element, IEnumerable<string
             return null;
         }
 
+        [Description("Searches a Revit Family Instance for the first existing parameter of a given name, by first looking for Instance parameters and then, optionally, for Type parameters.")]
+        [Input("element", "Revit element to be queried.")]
+        [Input("parameterName", "Names of the parameter to be iterated over in search for the parameter.")]
+        [Input("allowTypeParameters", "Optional, whether or not to also look for Type parameters if no Instance parameters were found.")]
+        [Output("parameter", "Parameter extracted from the input Revit Family Instance.")]
+        public static Parameter LookupParameter(this Element element, string parameterName, bool allowTypeParameters = true)
+        {
+            if (element == null || string.IsNullOrEmpty(parameterName))
+                return null;
+           
+            // Try to return an Instance Parameter
+            foreach (Parameter p in element.Parameters)
+            {
+                if (p != null && p.Definition.Name == parameterName)
+                    return p;
+            }
```

Comment:
I believe calling API methods is more efficient than iterating manually:
```suggestion
            Parameter parameter = element.LookupParameter("parameterName");
            if (parameter != null)
                return parameter;
```